src/btree/bt_slvg.c

```diff
@@ -612,7 +613,10 @@ __slvg_trk_leaf(WT_SESSION_IMPL *session,
 		 * on every leaf page, and if you need to speed up the salvage,
 		 * it's probably a great place to start.
 		 */
-		WT_ERR(__wt_page_inmem(session, NULL, dsk, 0, &page));
+		WT_CLEAR(tmp);
+		tmp.data = dsk;
+		tmp.memsize = dsk->mem_size;
```

Comment:
@agorrod, this scares me a lot -- setting the WT_ITEM's memsize, some code is going to look at that some time and think the WT_ITEM has associated memory. I'd go back to passing in the image and a length.